main.cpp

```diff
@@ -79,12 +84,34 @@ struct Temporary
         std::cout << "I'm a Temporary<" << typeid(v).name() << "> object, #"
                   << counter++ << std::endl;
     }
-    
+//Rule of 3
+    //Temporary(const Temporary& other) : v(other.v){}
+
+    // Temporary& operator= (const Temporary& other)
+    // {
+    //     v = other.v;
+    //     return *this;
+    // }
```

Comment:
If you want to know why you have to comment this out to make it compile, take a look at the definition of `JUCE_DECLARE_NON_COPYABLE_WITH_LEAK_DETECTOR.
- [ ] investigate.